JsonSubTypes/JsonSubtypes.cs

```diff
@@ -241,11 +241,16 @@ private Type GetType(JObject jObject, Type parentType, JsonSerializer serializer
             Type targetType = parentType;
             JsonSubtypes lastTypeResolver = null;
             JsonSubtypes currentTypeResolver = this;
+            HashSet<Type> visitedTypes = new HashSet<Type> {targetType};
 
             var jsonConverterCollection = serializer.Converters.OfType<JsonSubtypesConverter>().ToList();
             while (currentTypeResolver != null && currentTypeResolver != lastTypeResolver)
             {
                 targetType = currentTypeResolver.GetType(jObject, targetType);
+                if (!visitedTypes.Add(targetType))
```

Comment:
If you exit the loop a `JsonSerializationException` exception with the folowing message will be thrown :
`Could not create an instance of type JsonSubTypes.Tests.AbstractBaseClassDiscriminatorTests+MainClass. Type is an interface or abstract class and cannot be instantiated. Path 'Discriminator', line 1, position 17.`

I prefer the later

Reply:
If we keep the implementation with a hash set of visited types, I'll change it to just break out of the loop.